src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -143,13 +146,25 @@ private void connect(String username, String password, String hostname, int port
 
     try {
       ExperimenterData user = this.gateway.connect(loginCredentials);
-      this.securityContext = new SecurityContext(user.getGroupId());
+
+      long userGroupID = user.getGroupId();
+
+      if(!group.equalsIgnoreCase("default")){
+        for (int i = 0; i < user.getGroups().size(); i++) {
+          if(user.getGroups().get(i).getName().equalsIgnoreCase(group)){
+            userGroupID = user.getGroups().get(i).getGroupId();
+            break;
```

Comment:
I'D recommend to adapt this using [java streams](https://www.baeldung.com/java-stream-filter-lambda) which makes this more readable :) 

```suggestion
        for (int i = 0; i < user.getGroups().size(); i++) {
          if(user.getGroups().get(i).getName().equalsIgnoreCase(group)){
            userGroupID = user.getGroups().get(i).getGroupId();
            break;
```

Reply:
I will look into it